runtime/lua/vim/lsp.lua

```diff
@@ -1093,19 +1093,29 @@ function lsp.start_client(config)
     end
   end
 
+  ---@private
+  -- Determines whether the given option can be set by `set_defaults`.
+  local function is_empty_or_default(bufnr, option)
+    if vim.bo[bufnr][option] == '' then
+      return true
+    end
+    local last_set_from = vim.fn.gettext('\n\tLast set from ')
+    return not not vim.fn.execute('verbose set ' .. option .. '?'):match(last_set_from .. '.*ftplugin')
```

Comment:
I wonder  does there can not simply use `nvim_get_option_value` ?

Reply:
Nope, this does not provide information from where the option was set. I think that going through vimscript is currently the only way.

Reply:
it does, if you use `-V1`

Reply:
did you try use  lua file in`ftplugin`?  i guess this way might not works fine when it's lua file. you will get message like `last set from lua`

Reply:
@clason
Where should `-V1` go? This is not mentioned in the docs (https://neovim.io/doc/user/api.html).
The second parameter (opts) is either scope, buf or win.
```
:lua print(vim.api.nvim_get_option_value('omnifunc', {})) 
```

@glepnir
Yes it if is set from lua file it does work as you say. However, the goal of this PR is to overwrite only values set by builtin ftplugins shipped with neovim. All of them are implemented in vimscript right now. I think those are inherited from vim and will stay as vimscript files, right? You can find the list of impacted languages in the corresponding issue.

Reply:
No, I mean that if you run Nvim with `-V1`, you _do_ get information where the option was set from Lua. This is not sufficient justification to use vimscript here.

Reply:
A Vim patch may be relevant, combined with `nvim_get_option_info`: https://github.com/vim/vim/commit/f768c3d19c518822d89dec4cc3947ddeea249316